### DISCUSSION TITLE
How to plot "Rho" from Geometrically Necessary Dislocations calculation?

### DISCUSSION THREAD

**Discussion by imcrashed** [mtex-toolbox/mtex#921] [Ask Anything]
Hello everyone,

I just started using mtex because I wanted to do GND calculation for a deformed sample. I followed the tutorial "https://mtex-toolbox.github.io/GND.html" but in the end GND was plotted for different curvature components. I am not familiar with the curvature components. I want to have an overall GND plot (like sum of "Rho" perhaps) that can give me the GND with color bar in SI units, just about the same way as the Dislocation Energy Map was calculated at the end of tutorial. Can anyone please help me?

Best,

Shahzad

**Comment by kilir** [comment]
Hi,
I think you are looking for the last plot of `factor*sum(abs(rho .* dSRot.u),2)`. It has units of m^-2.
Cheers,
Rüdiger

**Reply by imcrashed** [reply]
Hi,

Thank you for your kind reply. I was thinking the same that although the heading reads TOTAL DISLOCATION ENERGY, which is being obtained by multiplying rho with dSRot.u. If I do not multiply and instead use syntax  "factor*sum(abs(rho),2)" it gives rho value instead of energy? Is my understanding correct? 

Plot does not change much because dSRot.u is almost equals to 1. 

btw, what does 2 signify at the end of syntax?

**Reply by matheusGariglio** [reply]
Hi Shahzad,

I think you are correct. Actually, doing `factor*sum(abs(rho .* dSRot.u),2)`, what is obtained is the **Energy of Dislocations**, not the Dislocation Density, so the unit is **J/m^3**. Doing` factor*sum(abs(rho),2)`, you get the **dislocation density (1/m^2)**. But I would also like to verify this with @ralfHielscher or @kilir.

Btw, I think the energy of screw dislocations is not right on the web page (https://mtex-toolbox.github.io/GND.html). From many references, the ratio between edge and screw dislocations is actually (**Eedge = Escrew * (1-nu)**). See some of them below,

- Introduction to Dislocations Hull & Bacon - Pages 71 et 72 (Eq 4.22) 

> (http://research.iaun.ac.ir/pd/ebrahimzadeh/pdfs/UploadFile_6472.pdf)

- Muransky et al. (2019) - Equation 12

>  (https://www.sciencedirect.com/science/article/pii/S1359645419303209)

The 2 value is just to show that screw and edge dislocations have different energies, check the first reference for more info.

Cheers,
Matheus

**Reply by imcrashed** [reply]
Hello Matheus,

Thank you for the references. Much appreciated. 

I think the energy of dislocation is not that easy to set. Just as you mentioned there are a few papers detailing how to obtain the energies, but there there remain ambiguity as well. Which is why I want to plot dislocation density RHO instead of energy. I think for most material analysis purposes, RHO would suffice. 

Do let me know if you hear more about how to set the energy or if I am wrong about how I plotted RHO in formula above.

Thanks.

Best,
Shahzad.
